syntax = "proto3";

package ovms_ipc;

import "data_attributes.proto";

message EndpointRequest{
    oneof type{
        PredictRequest predict_request = 1;
        ModelMetadataRequest model_metadata_request = 2;
        ModelStatusRequest model_status_request = 3;
    }
}

message PredictRequest{
    message Data{
        DataAttributes attributes = 1;
        string input_name = 2;
        string shm_name = 3;
    }
    repeated Data inputs = 1;
    string return_socket_name = 2;
}

message ModelMetadataRequest{
    int32 requested_version = 1;
}

message ModelStatusRequest{
    int32 requested_version = 1;
}
